spack/packages/dla-future/package.py

```diff
@@ -191,11 +191,22 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     variant("ci-test", default=False, description="Build for CI (Advanced usage).")
     conflicts("~miniapps", when="+ci-test")
 
+    # ci-check-threads conflicts for cuda or rocm builds as the libraries create their own threads.
+    # enabled by default when +ci-test
     variant(
         "ci-check-threads",
         default=False,
         description="Check number of spawned threads in CI (Advanced usage).",
     )
+    variant(
+        "ci-check-threads",
+        default=True,
+        when="+ci-test",
+        description="Check number of spawned threads in CI (Advanced usage).",
+    )
+    conflicts("+ci-check-threads", when="+cuda")
+    conflicts("+ci-check-threads", when="+rocm")
+    conflicts("+ci-check-threads", when="platform=darwin")
```

Comment:
Minor, but maybe rather:
```suggestion
        when="+ci-test platform=linux",
        description="Check number of spawned threads in CI (Advanced usage).",
    )
    conflicts("+ci-check-threads", when="+cuda")
    conflicts("+ci-check-threads", when="+rocm")
```
?

Reply:
It doesn't handle the `~ci-test` case and it would require duplication to handle `+ci-test` and `~ci-test`.

Reply:
Just adding this here since I learned after this comment that it's possible to define a variant multiple times in a package: https://spack.readthedocs.io/en/latest/packaging_guide.html#overriding-variants.